Epilepsy/epilepsy_utils.py

```diff
@@ -0,0 +1,119 @@
+from collections import defaultdict
+from itertools import product
+from string import punctuation
+
+from pandas import DataFrame
+
+def offsets_to_token(left, right, offset_array, lemmas, punc=set(punctuation)):
+    """Calculate the offset from tag to token
+
+    Ripped off from the snorkel custom tagger.
+    Designed to get the offset where a given token is found so it can receive a custom entity tag.
+    (i.e. Gene or Chemical)
+
+    Keyword arguments
+    left - the start of the tag
+    right - the end of the tag
+    offset_array - array of offsets given by stanford corenlp 
+    lemmas - array of lemmas given by stanford corenlp
+    punc - a list of punctuation characters
+    """
+    token_start, token_end = None, None
+    for i, c in enumerate(offset_array):
+        if left >= c:
+            token_start = i
+        if c > right and token_end is None:
+            token_end = i
+            break
+    token_end = len(offset_array) - 1 if token_end is None else token_end
+    token_end = token_end - 1 if lemmas[token_end - 1] in punc else token_end
+    return range(token_start, token_end)
+
+
+
+class Tagger(object):
+    """Custom Tagger Class
+    This is a custom class that is designed to tag each relevant word in a given sentence
+    i.e if it see GAD then this tagger will give GAD a Gene tag.
+    """
+    
+    def __init__(self,file_name):
+        """ Initialize the tagger class
+
+        Keyword arguments:
+        self -- the class object
+        file_name -- the name of the file that contains the document annotations
+        """
+        self.open_file = open(file_name,"r")
+
+    
+    def __del__(self):
+        self.open_file.close()
+
+    def retrieve_document(self,pubmed_id):
```

Comment:
As discussed let's switch to a tidy TSV for storing annotations. This will give us lot's of flexibility -- the TSV can be read using pandas and easily imported into a database.

For more on the philosophy of tidy data, see [this treatise on the matter](https://doi.org/10.18637/jss.v059.i10 "Hadley Wickham. Tidy Data. JSS. 2014").